### PR TITLE
respect pathPrefix

### PR DESCRIPTION
**Pull request by StrongMonkey**

### CODE REVIEW

**Comment by StrongMonkey** [comment]
https://github.com/rancher/rancher/issues/14376

File: pkg/controllers/management/node/controller.go [hunk 1]
```diff
@@ -254,11 +255,30 @@ func (m *Lifecycle) deployAgent(nodeDir string, obj *v3.Node) error {
 	if err != nil {
 		return err
 	}
+	prefix := "/"
+	cluster, err := m.clusterLister.Get("", obj.Namespace)
+	if err != nil && !kerror.IsNotFound(err) {
+		return err
+	}
```
Comment (alena1108) [inline pkg/controllers/management/node/controller.go]: if err is not found, then the cluster will be nil? We should just return nil if not found

File: pkg/rkenodeconfigserver/nodeserver.go [hunk 2]
```diff
@@ -111,6 +113,10 @@ func (n *RKENodeConfigServer) nonWorkerConfig(ctx context.Context, cluster *v3.C
 	if rkeConfig == nil {
 		rkeConfig = &v3.RancherKubernetesEngineConfig{}
 	}
+	prefix := cluster.Spec.RancherKubernetesEngineConfig.PrefixPath
```
Comment (alena1108) [inline pkg/rkenodeconfigserver/nodeserver.go]: can we extract this block - lines 116-119 - to its own method? it gets repeated a couple of times

File: pkg/controllers/management/node/controller.go [hunk 3]
```diff
@@ -254,11 +255,30 @@ func (m *Lifecycle) deployAgent(nodeDir string, obj *v3.Node) error {
...
+	if cluster.Spec.RancherKubernetesEngineConfig.PrefixPath != "" {
```
Comment (alena1108) [inline pkg/controllers/management/node/controller.go]: Nit: can we put prefix initialization close to the place where it's used - line 277?

File: pkg/controllers/management/node/controller.go (hunk 1 above)
Comment (StrongMonkey) [reply pkg/controllers/management/node/controller.go]: don' t think if cluster can't be found cause it must have one. If not we just default pathFix to "/"

File: pkg/controllers/management/node/controller.go (hunk 3 above)
Comment (StrongMonkey) [reply pkg/controllers/management/node/controller.go]: Line 258-275 is to get two thing(cluster.prefix and info.osType) so the order shouldn't matter :)

**Pull request merged**